internal/terminal/table.go

```diff
@@ -0,0 +1,131 @@
+package terminal
+
+import (
+	"errors"
+	"fmt"
+	"github.com/fatih/color"
+	"strings"
+)
+
+type table struct {
+	headers      []string
+	data         []map[string]string
+	columnWidths map[string]int
+}
+
+const (
+	// gutterLength is to set a standard for the break between columns for the table
+	gutterLength = 2
+)
+
+func parseInterface(data interface{}) string {
+	var normalizedData string
+	switch v := data.(type) {
+	case nil: // leave zero data
+		normalizedData = ""
+	case fmt.Stringer:
+		normalizedData = v.String()
+	default:
+		normalizedData = fmt.Sprintf("%v", v)
+	}
+	return normalizedData
+}
+
+func newTable(headers []string, data []map[string]interface{}) table {
+	resultTable := table{
+		[]string{},
+		[]map[string]string{},
+		map[string]int{},
+	}
+
+	if len(headers) == 0 {
+		return resultTable
+	}
+	resultTable.headers = headers
+
+	for _, header := range headers {
+		if headerLen := len(header); headerLen > resultTable.columnWidths[header] {
+			resultTable.columnWidths[header] = headerLen
+		}
+	}
+
+	for _, row := range data {
+		normalizedRow := make(map[string]string)
+		for key, rowData := range row {
+			// convert the interface data into strings for printing
+			normalizedRowData := parseInterface(rowData)
+			if rowDataLen := len(normalizedRowData); rowDataLen > resultTable.columnWidths[key] {
+				resultTable.columnWidths[key] = rowDataLen
+			}
+			normalizedRow[key] = normalizedRowData
+		}
+		resultTable.data = append(resultTable.data, normalizedRow)
+	}
+	return resultTable
+}
+
+func (t table) writeHeader() string {
+	var result []string
+	for _, header := range t.headers {
+		spaces := strings.Repeat(" ", t.columnWidths[header]-len(header))
+		result = append(result, fmt.Sprintf("%s"+spaces, color.New(color.Bold).SprintFunc()(header)))
+	}
+	return strings.Join(result, strings.Repeat(" ", gutterLength))
+}
+
+func (t table) writeRows() string {
+	var result []string
+	for _, dataMap := range t.data {
+		// if there's no data for that row, then ignore
+		if len(dataMap) > 0 {
+			var rowResult []string
+			for _, header := range t.headers {
+				spaces := strings.Repeat(" ", t.columnWidths[header]-len(dataMap[header]))
+				rowResult = append(rowResult, fmt.Sprintf("%s"+spaces, dataMap[header]))
+			}
+			result = append(result, strings.Join(rowResult, strings.Repeat(" ", gutterLength)))
+		}
+	}
+	return strings.Join(result, "\n")
+}
+
+func (t table) writeDivider() string {
+	var divider []string
+	for _, header := range t.headers {
+		divider = append(divider, strings.Repeat("-", t.columnWidths[header]))
+	}
+	return strings.Join(divider, strings.Repeat(" ", gutterLength))
+}
+
+const (
+	logFieldHeaders = "headers"
+	logFieldData    = "data"
+)
+
+func (t table) validate() error {
+	if len(t.headers) == 0 {
+		return errors.New("cannot create a table without headers")
+	}
+	return nil
+}
+
+func (t table) Message() (string, error) {
```

Comment:
Should this method be called `Print` since it is printing the table? Also can we get some comments on exported methods similar to what we do in the server code (Usually the linter picks this up there)?